File: Java/MergeLL.java
// Merge two sorted linked lists into one sorted list

class ListNode{
    int val;
    ListNode next;
    ListNode(int x) {val = x;}
}

public class MergeLL {
    static ListNode mergeLL(ListNode head1, ListNode head2) {
        ListNode tmp = new ListNode(0);
        ListNode cur = tmp;

        while(head1 != null && head2 != null){
            if(head1.val <= head2.val){
                cur.next = head1;
                head1 = head1.next;
            }
            else{
                cur.next = head2;
                head2 = head2.next;
            }
            cur = cur.next;
        }

        if(head1 != null) {cur.next = head1;}
        else if(head2 != null) {cur.next = head2;}
        return tmp.next;
    }

    static void printList(ListNode head){
        while(head != null){
            System.out.print(head.val + " ");
            head = head.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        ListNode head1 = new ListNode(1);
        head1.next = new ListNode(3);
        head1.next.next = new ListNode(5);
        ListNode head2 = new ListNode(2);
        head2.next = new ListNode(4);
        head2.next.next = new ListNode(6);

        System.out.println("Originals: ");
        printList(head1);
        printList(head2);

        ListNode Result = mergeLL(head1, head2);

        System.out.print("Merged: ");
        printList(Result);
    }
}

File: Python/MergeLL.py
#Merge two sorted linked lists into one sorted list

class ListNode:
    def __init__(self, val = 0, next = None):
        self.val = val
        self.next = next

def MergeLL(head1, head2):
    tmp = ListNode(0)
    cur = tmp

    while head1 is not None and head2 is not None:
        if head1.val <= head2.val:
            cur.next = head1
            head1 = head1.next
        else:
            cur.next = head2
            head2 = head2.next
        cur = cur.next

    if head1 is not None:
        cur.next = head1
    elif head2 is not None:
        return tmp.next 

def printList(head):
    while head:
        print(head.val, end=" ")
        head = head.next
    print()

def main():
    head1 = ListNode(1, ListNode(3, ListNode(5)))
    head2 = ListNode(2, ListNode(4, ListNode(6)))

    print("Originals:", end=" ")
    printList(head1)
    printList(head2)

    Result = MergeLL(head1, head2)

    print("Merged:", end=" ")
    printList(Result)

if __name__ == "__main__":
    main()